src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -867,6 +867,92 @@ internal sealed partial class LazyBinaryOperatorExpression : BaseBinaryOperatorE
         protected override IOperation RightOperandImpl => _lazyRightOperand.Value;
     }
 
+    /// <summary>
+    /// Represents an operation with two operands that produces a result with the same type as at least one of the operands.
+    /// </summary>
+    internal abstract class BaseTupleBinaryOperatorExpression : Operation, ITupleBinaryOperation
+    {
+        public BaseTupleBinaryOperatorExpression(BinaryOperatorKind operatorKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit)
+            : base(OperationKind.BinaryOperator, semanticModel, syntax, type, constantValue, isImplicit)
+        {
+            OperatorKind = operatorKind;
+        }
+        /// <summary>
+        /// Kind of binary operation.
+        /// </summary>
+        public BinaryOperatorKind OperatorKind { get; }
+        /// <summary>
+        /// Left operand.
+        /// </summary>
+        public IOperation LeftOperand => Operation.SetParentOperation(LeftOperandImpl, this);
```

Comment:
>Operation.SetParentOperation(LeftOperandImpl, this); [](start = 41, length = 52)

We are moving away from the pattern when parent is always set lazily. Only "lazy" sub-class should do this, non-lazy subclass should assign parent for children in constructor.

Reply:
I was mirroring binary expressions. Which node is the correct/new pattern?